crud.js

```diff
@@ -0,0 +1,157 @@
+//Has functions to CREATE READ UPDATE and DELETE
+module.exports =
+  function(database, password, user, host, port) {
+    assert(database !== undefined, "No Database Specified");
+    assert(password !== undefined, "Require Password");
+    (user !== undefined) ? user = user: user = "root";
+    (host !== undefined) ? host = host: host = "localhost";
+    (port !== undefined) ? port = port: port = 3306;
+    this.connection = mysql.createConnection({
+      host: host,
+      port: port,
+      user: user,
+      password: password,
+      database: database
+    });
+
+    //Creates a new entry in the database
+    //table is the name of the table or an array of table names
+    //set is an object containing the values to set for hte table
+    //callback function called when CREATE is done
+    this.CREATE = function(table, set, callback) {
+      assert(table !== undefined || set !== undefined || callback !== undefined,
+        "Specicify table and set and add a listener");
+      var query = "INSERT INTO ?? SET ?";
+      var call = this.connection.query(query, [table, set], function(err, res) {
+        if (err) {
+          throw err;
+        }
+        callback(res)
+      });
+    };
+
+    //Reads tables specified in from
+    //From is the name of the table or and array of tables
+    //options is an object of optional parameters
+    //listener is a callback function to be performed after everything is done in the READ
+    //options can be:
+    //  select: the columns or columns to read
+    //  where: an object where keys are columns where the value is true or
+    //    a string of arguments where values for columns are true
+    //  having: a string of arguments where values for columns are true
+    //  orderBy: 
+    //  limit:
+    //  query: if true will composite a mySQL query with the select command and what 
+    //    is written in from. Does not allow for multiple commands. 
+    this.READ = function(from, listener) {
+      assert(arguments.length >= 2 && typeof arguments[arguments.length - 1] === "function",
+        "Specicify FROM and a callback function");
+      var select, where, groupBy, having, orderBy, limit, command;
+      var params = [];
+      if (typeof arguments[1] == "object") {
+        assert(typeof arguments[2] === "function", "requires callback function");
+        select = arguments[1].select;
+        where = arguments[1].where;
+        groupBy = arguments[1].groupBy;
+        having = arguments[1].having;
+        orderBy = arguments[1].orderBy;
+        limit = arguments[1].limit;
+        command = arguments[1].query;
+        listener = arguments[2];
+      } else {
+        assert(typeof arguments[1] === "function", "requires callback function");
+        listener = arguments[1];
+      }
+      if (command === true) {
+        var query = "SELECT " + from;
+      } else {
+        if (select === undefined) select = "*";
+        params.push(select);
+        params.push(from);
```

Comment:
I think you need thes lines contained within the above if otherwise it seems like it will get overwritten by what is below it.